### PR TITLE
Protobuf GitHub Action Workflow

### PR DESCRIPTION
**Pull request by scotwells**
This GitHub Action workflow will run the [API linter](https://linter.aip.dev) to ensure all protobuf definitions are compliant with [AIP standards](https://google.aip.dev/general) and generate the Protobuf code to ensure no changes are detected. A GitHub Action summary will be outputted for the build with any detected file changes. See the build summaries in #66 for an example.

This also includes some other fixes:

- Changes gomodule name for generated Protobuf code to `go.datumapis.com/os/genproto` so its namespaced from datum-cloud.
- Changes `google.api.FieldMask` to `google.protobuf.FieldMask` to resolve Protobuf compilation issues
- Imports missing Protobuf files to resolve Protobuf generation errors
- Adds missing `go_package` annotation to resolve Protobuf generation errors
- Installs missing `go install github.com/google/gnostic/cmd/protoc-gen-openapi@latest` command to generate openapi specification
- Adds operating system dependent task files to install Protobuf correctly based on the user's operating system.

### CODE REVIEW

File: api/datum/os/vendormanager/v1/folders_service.proto
Comment (scotwells) [inline api/datum/os/vendormanager/v1/folders_service.proto:1]: @nullaus looks like the file name on this should have been `vendors_service.proto`? Do you want to change that here or another PR? What are your thoughts on using `v1alpha` as the version of the service until we consider the API stable?

**Comment by scotwells** [comment]
@nullaus I noticed the GitHub action is complaining that `api/openapi/openapi.yaml` is being generated but isn't committed to the repo. Was that intentional? 

There's also a few linter violations that need to be addressed on the vendor manager service. Do you want me to resolve those on this PR?

File: api/datum/os/vendormanager/v1/folders_service.proto
Comment (nullaus) [reply api/datum/os/vendormanager/v1/folders_service.proto:1]: Ah yeah, it's renamed in https://github.com/datum-cloud/datum-os/pull/64. I'll have a think on `v1alpha` vs `v1`. I think I might be more inclined to use something like `v0` to show that it's not a stable API yet.

Comment (scotwells) [reply api/datum/os/vendormanager/v1/folders_service.proto:1]: We've been following the [channel-based versioning](https://cloud.google.com/apis/design/versioning#channel-based_versioning) guidance proposed by Google on the datum-cloud side. 

That guidance recommends `v1` would represent the stable version of the `v1` API, `v1beta` would represent the beta version of the `v1` API, and `v1alpha` would represent the alpha version of the `v1` API.

Graduating an API from unstable to stable would typically go `v1alpha -> v1beta -> v1`. If we wanted to introduce a new v2 version, we'd start with `v2alpha` and graduate towards a `v2` API.

**Review by nullaus — APPROVED** [review]
Looks good, thank you!

**Pull request merged**